packages/kokkos-kernels/unit_test/blas/Test_Blas3_gemm.hpp

```diff
@@ -105,9 +105,10 @@ namespace Test {
     uint64_t seed = Kokkos::Impl::clock_tic();
     Kokkos::Random_XorShift64_Pool<execution_space> rand_pool(seed);
 
-    Kokkos::fill_random(A,rand_pool,ScalarA(10));
-    Kokkos::fill_random(B,rand_pool,ScalarB(10));
-    Kokkos::fill_random(C,rand_pool,ScalarC(10));
+    // (SA 11 Dec 2019) Max (previously: 10) increased to detect the bug in Trilinos issue #6418 
+    Kokkos::fill_random(A,rand_pool, Kokkos::rand<typename Kokkos::Random_XorShift64_Pool<execution_space>::generator_type,ScalarA>::max());
```

Comment:
How big is this `rand<>:::max()`?  Is this going to cause problems with overflow?

Reply:
Yes, the updated test fails if I revert the commit that updates the file:
    kokkos-kernels/src/impl/tpls/KokkosBlas3_gemm_tpl_spec_decl.hpp

I'm not sure about the value of rand<>::max(), but I want to say it should have caused problems in the Tpetra gemm test if it would (?). On the other hand, the inner dimension of the multiplication in the KokkosKernels test goes up to 517 whereas this value is only 13 in the Tpetra test, so KK test would be more likely to hit such an issue (?).  I'm not sure.